ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1372,15 +1424,24 @@ void handleBody(InputStream payload, MessageMetadata messageMetadata, MessageInf
           rawPayloadBuffer = ByteBuffer.wrap(payloadBytes);
         }
 
-        if (getOperationFlag() == MessageFormatFlags.Blob) {
+        if (getOperationFlag() == MessageFormatFlags.BlobInfo) {
+          blobType = BlobType.MetadataBlob;
+          blobData = null;
+          serverBlobProperties = MessageFormatRecord.deserializeBlobProperties(payload);
+          ByteBuffer userMd = MessageFormatRecord.deserializeUserMetadata(payload);
+          blobInfo = new BlobInfo(serverBlobProperties, userMd.array(), messageInfo.getLifeVersion());
```

Comment:
I think the user metadata in the metadata blob might be encrypted. We might have to use the user metadata/blobInfo from the metadata cache.